### ISSUE TITLE
Argument authorization called twice with pundit integration

### ISSUE THREAD

**Issue by Mathieuu** [rmosolgo/graphql-ruby#3242]
**Describe the bug**

While overriding `authorized?` in `Types::BaseArgument`, I noticed that the method is called twice if the mutation using the argument is including `GraphQL::Pro::PunditIntegration::MutationIntegration`. This is problematic because I am doing database calls in the `authorized?` method.

`lib/graphql/schema/resolver.rb:145` in the graphql gem calls it a first time:
```
--> #0  Types::BaseArgument.authorized?(*_args#Array, _kwargs#Hash) at /Users/backend/app/graphql/types/base_argument.rb:7
    #1  block in GraphQL::Schema::Resolver.block in authorized?(inputs#Hash) at /Users/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/graphql-1.11.6/lib/graphql/schema/resolver.rb:145
    ͱ-- #2  Hash.each_value at /Users/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/graphql-1.11.6/lib/graphql/schema/resolver.rb:142
    #3  GraphQL::Schema::Resolver.authorized?(inputs#Hash) at /Users/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/graphql-1.11.6/lib/graphql/schema/resolver.rb:142
    #4  GraphQL::Pro::PunditIntegration::AuthorizeMutationByPolicy.authorized?(inputs#Hash) at /Users/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/graphql-pro-1.16.0/lib/graphql/pro/pundit_integration.rb:203
```

then `graphql-pro-1.16.0/lib/graphql/pro/pundit_integration.rb:207` calls it a second time:

```
    --> #0  Types::BaseArgument.authorized?(*_args#Array, _kwargs#Hash) at /Users/backend/app/graphql/types/base_argument.rb:7
    #1  block in GraphQL::Pro::PunditIntegration::AuthorizeMutationByPolicy.block in authorized?(inputs#Hash) at /Users/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/graphql-pro-1.16.0/lib/graphql/pro/pundit_integration.rb:207
    ͱ-- #2  Hash.each at /Users/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/graphql-pro-1.16.0/lib/graphql/pro/pundit_integration.rb:203
    ͱ-- #3  Enumerable.all?(*args) at /Users/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/graphql-pro-1.16.0/lib/graphql/pro/pundit_integration.rb:203
    #4  GraphQL::Pro::PunditIntegration::AuthorizeMutationByPolicy.authorized?(inputs#Hash) at /Users/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/graphql-pro-1.16.0/lib/graphql/pro/pundit_integration.rb:203
```

**Versions**

`graphql` version: 1.11.6

**Steps to reproduce**

With a `base_mutation.rb`:

```rb
# frozen_string_literal: true

module Mutations
  class BaseMutation < GraphQL::Schema::RelayClassicMutation
    include GraphQL::Pro::PunditIntegration::MutationIntegration

    argument_class Types::BaseArgument
    field_class Types::BaseField
    input_object_class Types::BaseInputObject
    object_class Types::BaseObjectRecord
end
```

And 

```rb
module Types
  class BaseArgument < GraphQL::Schema::Argument
    def authorized?(*_args, **_kwargs)
      puts "Authorize? called for #{owner} #{keyword}"
      true
    end
  end
end
```

Any mutation using the base mutation will print twice `Authorize? called for #{owner} #{keyword}`.

**Expected behavior**

`authorized?` should only be called once.

**Comment by rmosolgo** [comment]
Hey, thanks for the detailed report! It looks like, when I upstreamed resolver argument authorization from graphql-pro to graphql-ruby, I didn't properly update the `super` calls in GraphQL-Pro 😖 . I removed that duplicate code from GraphQL-Pro and shipped 1.16.1 with a fix. Could you give it a try and let me know how it looks for you?

**Comment by Mathieuu** [comment]
Solved! Thank for the quick turnaround.
